internal/pkg/configuration/types.go

```diff
@@ -28,17 +33,21 @@ func (flags *stringArrayFlags) Set(value string) error {
 	return nil
 }
 
-// Valid node address types
+// NodeAddressType specifies valid node address types
 type NodeAddressType string
 
 const (
+	// NodeAddressTypeExternal is the constant for external node address types
 	NodeAddressTypeExternal = "external"
+	// NodeAddressTypeInternal is the constant for internal node address types
 	NodeAddressTypeInternal = "internal"
 )
 
 func (flags *NodeAddressType) String() string {
 	return string(*flags)
 }
+
+// Set is used for setting the node address type
```

Comment:
Set() and String() are actually requirements on the NodeAddressType to be a valid flag value (for https://github.com/cbeneke/hcloud-fip-controller/blob/d1f41fc648f6ddd3489c6e01ad274e363b2623dc/cmd/fip-controller/main.go#L22 )